src/liballoc/collections/btree/map.rs

```diff
@@ -692,6 +692,39 @@ impl<K: Ord, V> BTreeMap<K, V> {
         }
     }
 
+    /// Inserts a key-value pair into the map,
+    /// returning references to key and value.
+    pub fn insert_and_get_key_value(&mut self, key: K, value: V) -> (&K, &V) {
```

Comment:
This function seems redundant; the references returned from `_mut` can be downgraded to immutable at the callsite.

Reply:
The same argument could be applied against `get` as there is `get_mut`.

Reply:
Not quite, as those take `&self` and `&mut self`, respectively, while the new methods always take `&mut self`